update/Pos/Binary/Update.hs

```diff
@@ -118,35 +114,25 @@ deriveSimpleBi ''U.VoteState [
     Cons 'U.NegativeRevote []]
 
 instance Bi a => Bi (U.PrevValue a) where
-    sizeNPut = labelS "PrevValue" $
-        convertToSizeNPut $ \case
-            U.PrevValue v -> putWord8S 2 <> putS v
-            U.NoExist     -> putWord8S 3
-    get = label "PrevValue" $ getWord8 >>= \case
-        2 -> U.PrevValue <$> get
-        3 -> pure U.NoExist
-        x -> fail $ "get@PrevValue: invalid tag: " <> show x
+  encode (U.PrevValue a) = encodeListLen 2 <> encode (2 :: Word8) <> encode a
+  encode U.NoExist       = encodeListLen 1 <> encode (3 :: Word8)
+  decode = do
+    len <- decodeListLen
+    tag <- decode @Word8
+    case (len, tag) of
+      (2,2) -> U.PrevValue <$> decode
+      (1,3) -> pure U.NoExist
```

Comment:
It's very minor, but I think we can save 1 byte here and not store tag but deduce it from the length. That's how it's done for `Maybe`.